politeiad/backend/gitbe/cms.go

```diff
@@ -0,0 +1,1183 @@
+// Copyright (c) 2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package gitbe
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/decred/politeia/cmsplugin"
+	"github.com/decred/politeia/politeiad/api/v1/identity"
+	"github.com/decred/politeia/politeiad/backend"
+	"github.com/decred/politeia/util"
+)
+
+const (
+	cmsPluginIdentity  = "cmsfullidentity"
+	cmsPluginJournals  = "cmssjournals"
+	cmsPluginInventory = "cmsinventory"
+)
+
+type CastDCCVoteJournal struct {
+	CastVote cmsplugin.CastVote `json:"castvote"` // Client side vote
+	Receipt  string             `json:"receipt"`  // Signature of CastVote.Signature
+}
+
+func encodeCastDCCVoteJournal(cvj CastDCCVoteJournal) ([]byte, error) {
+	b, err := json.Marshal(cvj)
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
+func decodeCastDCCVoteJournal(payload []byte) (*CastDCCVoteJournal, error) {
+	var cvj CastDCCVoteJournal
+
+	err := json.Unmarshal(payload, &cvj)
+	if err != nil {
+		return nil, err
+	}
+
+	return &cvj, nil
+}
+
+var (
+	cmsPluginSettings map[string]string             // [key]setting
+	cmsPluginHooks    map[string]func(string) error // [key]func(token) error
+
+	// Cached values, requires lock. These caches are lazy loaded.
+	// XXX why is this a pointer? Convert if possible after investigating
```

Comment:
This was copied from the decred plugin but is a mistake.  There isn't any reason to make the StartVote a pointer so don't copy that part.